packages/grafana-data/src/utils/namedColorsPalette.ts

```diff
@@ -65,61 +67,59 @@ const buildColorDefinition = (
   isPrimary: !!isPrimary,
 });
 
-export const getColorDefinitionByName = (name: Color): ColorDefinition => {
+export function getColorDefinitionByName(name: Color): ColorDefinition {
```

Comment:
:heart: I am happy every time I see these constants replaced!

Reply:
Interesting, why `function` is preferable over `const` in this case (there are some arguments I've heard when creating funcs inside a Class)? I always try to follow the style of the existing functions but for green field files I always use `const`. I don't have strong opinions about this but maybe we should have a code style guideline for this so we keep the code consistent?

Reply:
It may just be personal taste :)  arrow function in classes/objects will bind `this` for you, so that is great.  

I find it less clear to define function as a const... when it is really intended to be a function.

Reply:
I find functions much better, they get hoisted to the top so order is not important. It's more readable and clear that you are looking at function and not a variable. 

It used to be that normal functions had better stack traces as the function had a name instead of being an anonymous function but not sure if this is the case anymore.

Reply:
Here is a short example of what happens after TypeScript transpiler: https://www.typescriptlang.org/play?#code/MYewdgzgLgBAZiEMC8MAUBKFA+GBvAWACgZSYAnAUygFdywYBGAbmIF9Wji4axgoAluBgAjAIblM+YmQrU6DFu2LFgAGzEQIMAGKIAQhOkkyCJKinJchLidIcVd0RKk3ZsmWTbsgA

Reply:
yes, that highlights why I never like arrow functions in classes as well :) 
unless you have to for a callback to bind this

Reply:
I prefer `function` over `const` and the primary reason is that with `const` syntax is hard(hacks needed) to create generic functions in `.tsx` files.

Reply:
Hacks needed?

Reply:
> with const syntax is hard(hacks needed) to create generic functions
@dprokop what do you mean by that?

I like arrow functions for higher order functions ie `() => () =>` but is a weak preference because as Torkel says the lack of hoisting can be problematic and the missing function names in stack traces.